vulture_os/authentication/openid/views.py

```diff
@@ -85,13 +88,28 @@ def edit(request, object_id=None, api=False):
         form = OpenIDRepositoryForm(request.POST or None, instance=repo, error_class=DivErrorList)
 
     if request.method in ("POST", "PUT") and form.is_valid():
+        old_jwt_key_filename = repo.get_jwt_key_filename()
         # Save the form to get an id if there is not already one
         repo = form.save(commit=False)
         # If provider_url changed, force reload of configuration
         if "provider_url" in form.changed_data:
             repo.last_config_time = None
+        if "jwt_key" in form.changed_data and repo.jwt_signature_type not in ("HS256", "HS384", "HS512"):
+            Cluster.api_request("system.config.models.delete_conf", old_jwt_key_filename)
+            repo.save_conf()
```

Comment:
Looks like 2 different situations and conditions to me:
1. need to delete the file if 
  - jwt_signature_type changed from HSxxx to something else
  - (OpenID is deleted)
  - OpenID name is modified (as name is included in filename)
2. need to update existing configuration file if
  - name is changed (see above)
  - jwt_key is changed

Reply:
The conditions above are now taken into account